server/metrics/global_status.go

```diff
@@ -29,24 +29,28 @@ var ignoredFieldsForWrites = [][]byte{
 	[]byte("_tigris_updated_at"),
 }
 
+type UsageProvider interface {
+	Flush() TenantStatusTimeChunk
+}
+
 type GlobalStatus struct {
 	mu          sync.Mutex
 	activeChunk *TenantStatusTimeChunk
 }
 
 type TenantStatusTimeChunk struct {
-	startTime time.Time
+	StartTime time.Time
```

Comment:
The idea is that these can only be written with a lock. Making these public loses that.

Reply:
Another option I can think of is `Flush()` returning a copy instead of the original object

Reply:
Flush is returning a copy and zeroing out the original object while holding a lock. So if a request finishes in the meantime, we log the stats for those in the new structure, and the reporter will get it in the next flush.

Reply:
There can be an interface here(`TenantTimeChunk`) that has getters, and then that interface can be mocked easily in the reporter.